pyclaragenomics/setup_pyclaragenomics.py

```diff
@@ -75,14 +95,33 @@ def build(self):
         self.run_build_cmd()
 
 
-def setup_python_binding(is_develop_mode, pycga_dir, cga_install_dir):
-    subprocess.check_call(['pip', 'install'] + (['-e'] if is_develop_mode else []) + ["."],
+def setup_python_binding(is_develop_mode, wheel_output_folder, pycga_dir, cga_install_dir):
+    if wheel_output_folder:
+        # Copies shared libraries into clargenomics package
+        copy_all_files_in_directory(
+            os.path.join(cga_install_dir, "lib"),
+            os.path.join(pycga_dir, "claragenomics/shared_libs/"),
+        )
+        setup_command = ['python', 'setup.py', 'bdist_wheel', '-d', wheel_output_folder]
+        completion_message = \
+            "A wheel file was create for pyclaragenomics under {}".format(wheel_output_folder)
+        cga_runtime_lib_dir = os.path.join('$ORIGIN', os.pardir, 'shared_libs')
```

Comment:
I'm wondering now, for the case of simplicity, why don't we use this setup for both cases? i.e. copy the shared libs over into the shared_libs folder, and then call either pip or python setup bdist. the way we have one less path setting to maintain. and we can add a .gitignore file which masks the shared_libs folder under pyclaragenomics so those copied files don't show up in the git index. what do you think?

Reply:
this will of course require the user to copy over the libs into the shared_libs folder location if they wanted to call `setup.py` by itself (without using our wrapper). I think if that case comes up (I expect it won't anytime soon) we can re-evaluate. We should just push for our users to run the wrapper script and not worry about calling things directly

Reply:
or can the shared lib copying be moved into the `setup.py` script which copes files from he environment variable location?

Reply:
@tijyojwad, the reason I didn't copy the files when installing through 'pip install' is that setuptools setup() ignores the 'package_data'  field when a package is not created (I can try to use data_files instead). Also, I already distinguish between 'pip install' and 'setup bdist_wheel' in setup_pyclaragenomics, so I did not want to do this again in setup.py

Reply:
@tijyojwad  You are right about copying the shared libraries copy setup.py, it makes more sense.

Reply:
@tijyojwad, @vellamike, I'm also considering using 'pip wheel' instead of 'python setup.py bdist_wheel'.
The wheel package leftovers- 'build/'  & 'pyclaragenomics.egg-info/'  folders under pyclaragenomics cause an unexpected behaviour if not deleted in the following scenario:
1. run `python setup_pyclaragenomics.py  --create_wheel_only`
2. run `python setup_pyclaragenomics.py `
now examine the `rpath` of the binding so files using `objdump` command and you can see they point to `$ORIGIN/../shared_libs` which is the `rpath` of the wheel creation int he previous command. pip perform this in a temporary folder under `/tmp` which does not leave leftovers under the repository folder.